backend.go

```diff
@@ -22,42 +54,49 @@ func Factory(ctx context.Context, conf *logical.BackendConfig) (logical.Backend,
 }
 
 func Backend(client ldapClient) *backend {
-	var b backend
+	b := &backend{
+		client:            client,
+		credRotationQueue: queue.New(),
+		roleLocks:         locksutil.CreateLocks(),
+		checkOutLocks:     locksutil.CreateLocks(),
+		managedUsers:      make(map[string]struct{}),
+	}
 
 	b.Backend = &framework.Backend{
 		Help: strings.TrimSpace(backendHelp),
-
 		PathsSpecial: &logical.Paths{
 			LocalStorage: []string{
 				framework.WALPrefix,
 			},
 			SealWrapStorage: []string{
-				"config",
-				"static-role/*",
+				configPath,
+				staticRolePath + "*",
 			},
 		},
 		Paths: framework.PathAppend(
-			b.pathListStaticRoles(),
+			b.pathConfig(),
+			b.pathDynamicRoles(),
+			b.pathDynamicCredsCreate(),
 			b.pathStaticRoles(),
 			b.pathStaticCredsCreate(),
 			b.pathRotateCredentials(),
-			b.pathConfig(),
-
-			b.pathDynamicRoles(),
+			b.pathSetCheckIn(),
+			b.pathSetManageCheckIn(),
+			b.pathSetCheckOut(),
+			b.pathSetStatus(),
+			b.pathSets(),
+			b.pathListSets(),
```

Comment:
This is copied from [vault-plugin-secrets-ad](https://github.com/hashicorp/vault-plugin-secrets-ad).